ipaserver/install/kra.py

```diff
@@ -69,9 +68,19 @@ def install_check(api, replica_config, options):
 
 
 def install(api, replica_config, options):
-    if replica_config is None:
-        if not options.setup_kra:
-            return
+
+    if replica_config is not None and not replica_config.setup_kra:
+        return
+
+    install_kra_replica = False
+    kra = krainstance.KRAInstance(api.env.realm)
+
+    if kra.is_installed() and replica_config is not None:
+        install_kra_replica = False
+    elif not kra.is_installed():
+        install_kra_replica = True
+
```

Comment:
I think you can completely remove the install_kra_as_replica "middleman" variable and directly check only the api.Command below here.